sources/Inputs/src/Node.cpp

```diff
@@ -19,7 +19,19 @@
 
 namespace dfl {
 namespace inputs {
-Node::Node(const NodeId& idNode, double nominalVoltageNode) : id(idNode), nominalVoltage{nominalVoltageNode}, neighbours{} {}
+
+std::shared_ptr<Node>
+Node::build(const NodeId& id, const std::shared_ptr<VoltageLevel>& vl, double nominalVoltage) {
+  auto ret = std::shared_ptr<Node>(new Node(id, vl, nominalVoltage));
```

Comment:
Why not use std::make_shared ?

Reply:
std::make_shared needs the constructor to be public (maybe in higher standard it is not the case)